openmp/libomptarget/src/omptarget.cpp

```diff
@@ -199,7 +199,7 @@ static int initLibrary(DeviceTy &Device) {
                                         Entry.size) != OFFLOAD_SUCCESS)
               REPORT("Failed to write symbol for USM %s\n", Entry.name);
           }
-        } else {
+        } else if (Entry.addr) {
```

Comment:
So now we don't have a "default" else branch here. Is it not needed?

Reply:
Also, could you explain in a comment why Entry.addr is used here?

Reply:
This is just hacking around the offloading entires. I wanted to use the same struct and handling for convenience, but right now there's no basic flag for what the kind of the thing is. Previously we just had functions and globals, if the size is zero it's a global. I added a third thing which can also be checked if the address is nullptr currently. I want to overhaul this in the future but I'm hoping to get this stuff working first.

Reply:
If the address is not-null, then it's a kernel waiting to be registered. It's a little hacky but in the current construction a kernel will always have a stub address pass in by the compiler. So one of these "requires" entries will have the addr be nullptr and the size be zero.

Reply:
Should we check for size > 0 too?

Reply:
Or is that too restrictive

Reply:
Isn't that already done in the `if` above?

Reply:
So it will only enter this branch if size is 0 and then if the address is not nullptr.

Reply:
Yes, I'll make the offloading entries less dumb in the future. I don't like how for working w/ CUDA and stuff we need `cuda_offloading_entries` and  whatnot as well. I've been planning on merging them and simply having a field for the language kind and stuff. Been planning on doing that for awhile.